modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -1309,6 +1309,227 @@ CV__DNN_INLINE_NS_BEGIN
+     * @param[in] network Net object
+     */
+    CV_WRAP TextRecognitionModel(const Net& network);
+
+    /**
+     * @brief Create text recognition model from network represented in one of the supported formats
+     * Call setDecodeType() and setVocabulary() after constructor to initialize the decoding method
+     * @param[in] model Binary file contains trained weights
+     * @param[in] config Text file contains network configuration
+     */
+    CV_WRAP inline
+    TextRecognitionModel(const std::string& model, const std::string& config = "")
+        : TextRecognitionModel(readNet(model, config)) { /* nothing */ }
+
+    /**
+     * @brief Set the decoding method of translating the network output into string
+     * @param[in] decodeType The decoding method of translating the network output into string: {'CTC-greedy': greedy decoding for the output of CTC-based methods}
+     */
+    CV_WRAP
+    TextRecognitionModel& setDecodeType(const std::string& decodeType);
+
+    /**
+     * @brief Get the decoding method
+     * @return the decoding method
+     */
+    CV_WRAP
+    const std::string& getDecodeType() const;
+
+    /**
+     * @brief Set the vocabulary for recognition.
+     * @param[in] vocabulary the associated vocabulary of the network.
+     */
+    CV_WRAP
+    TextRecognitionModel& setVocabulary(const std::vector<std::string>& vocabulary);
+
+    /**
+     * @brief Get the vocabulary for recognition.
+     * @return vocabulary the associated vocabulary
+     */
+    CV_WRAP
+    const std::vector<std::string>& getVocabulary() const;
+
+    /**
+     * @brief Given the @p input frame, create input blob, run net and return recognition result
+     * @param[in] frame The input image
+     * @return The text recognition result
+     */
+    CV_WRAP
+    std::string recognize(InputArray frame) const;
+
+    /**
+     * @brief Given the @p input frame, create input blob, run net and return recognition result
+     * @param[in] frame The input image
+     * @param[in] roiRects List of text detection regions of interest (cv::Rect, CV_32SC4). ROIs is be cropped as the network inputs
+     * @param[out] results A set of text recognition results.
+     */
+    CV_WRAP
+    void recognize(InputArray frame, InputArrayOfArrays roiRects, CV_OUT std::vector<std::string>& results) const;
+};
+
+
+/** @brief Base class for text detection networks
+ */
+class CV_EXPORTS_W TextDetectionModel : public Model
+{
+protected:
+    CV_DEPRECATED_EXTERNAL  // avoid using in C++ code, will be moved to "protected" (need to fix bindings first)
+    TextDetectionModel();
+
+public:
+    /** @brief Performs detection
+     *
+     * Given the input @p frame, prepare network input, run network inference, post-process network output and return result detections.
+     *
+     * @param[in] frame the input image
+     * @return array with text detection results
+     */
+    CV_WRAP
+    std::vector<cv::RotatedRect> detect(InputArray frame) const;
```

Comment:
Actually `RotatedRect` doesn't work well with perspective transformations (like [this one](https://www.pngwing.com/en/free-png-pdprp)).

Perhaps we need 4 points in API with strong order (bl, tl, tr, br - according to `targetVertices`) which should be used with `getPerspectiveTransform()` to get more accurate results.

I will try to update API on this week.

Reply:
I can not open the above link. Can you share it with google drive?

Reply:
It is similar to side boxes of cube from [here](https://en.wikipedia.org/wiki/Perspective_(graphical)#/media/File:Perspective1.jpg).

Reply:
However, DL models may not work with perspective transformations (EAST output doesn't know anything about that) and they just detect rotated text.

any thoughts?

Reply:
It depends on different DL models. Some methods can output irregular quadrilaterals.

From my side, the perspective transformation is a temporary replacement, or just a choice for the four-point outputs. (rotated boxes and irregular quadrilaterals)

There is a popular and fast text recognition algorithm `ASTER` which adopts `Thin Plate Spline Transformation` in its rectification network. (like [this](https://github.com/ayumiymk/aster.pytorch/blob/master/overview.png)).
I am not sure whether the TPS transformation is implemented in OpenCV, maybe not?
You can refer to [this](https://github.com/ayumiymk/aster.pytorch/blob/be670046c775b54de79766208f0c59321ae1eccf/lib/models/tps_spatial_transformer.py#L98).

By the way, ASTER is also an algorithm from our lab, and we are glad to contribute it to OpenCV.
However, there are some things to do before.
1) support TPS transformation
2) update LSTM in modules/dnn/src/onnx/onnx_importer.cpp
we need to set [these parameters](https://github.com/opencv/opencv/blob/ef0eed8d3c63b64bacc755dc4ffe5358a2cf30a7/modules/dnn/src/onnx/onnx_importer.cpp#L904) non-zero, but it is not supported now.
3) support [GRU layer](https://pytorch.org/docs/stable/generated/torch.nn.GRU.html?highlight=gru#torch.nn.GRU)
4) ...

I suppose to work on it after this PR.